jax/lax/lax.py

```diff
@@ -2230,70 +2306,96 @@ def _conv_general_dilated_transpose_rhs(
     out = conv_general_dilated(
       new_lhs, new_rhs, window_strides, padding, lhs_dilation, rhs_dilation,
-      dimension_numbers,
-      feature_group_count=lhs.shape[lhs_bdim] * feature_group_count,
+      dimension_numbers, feature_group_count=feature_group_count,
+      batch_group_count=batch_group_count,
       precision=precision)
     out = _reshape_axis_out_of(out_spec[1], lhs.shape[lhs_bdim], out)
     return out, out_spec[1]
 
   elif lhs_bdim is not None:
-    new_lhs = _reshape_axis_into(lhs_bdim, lhs_spec[0], lhs)
-    out = conv_general_dilated(new_lhs, rhs, window_strides, padding,
-                               lhs_dilation, rhs_dilation, dimension_numbers,
-                               feature_group_count, precision=precision)
-    out = _reshape_axis_out_of(out_spec[0], lhs.shape[lhs_bdim], out)
-    return out, out_spec[0]
+    if batch_group_count == 1:
+      new_lhs = _reshape_axis_into(lhs_bdim, lhs_spec[0], lhs)
+      out = conv_general_dilated(new_lhs, rhs, window_strides, padding,
+                                 lhs_dilation, rhs_dilation, dimension_numbers,
+                                 feature_group_count, precision=precision)
+      out = _reshape_axis_out_of(out_spec[0], lhs.shape[lhs_bdim], out)
+      return out, out_spec[0]
+    else:
+      new_lhs = _reshape_axis_out_of(lhs_spec[0] + int(lhs_bdim <= lhs_spec[0]),
+                                     batch_group_count, lhs)
+      new_lhs = _reshape_axis_into(lhs_bdim + int(lhs_spec[0] < lhs_bdim),
+                                   lhs_spec[0] + 1,
+                                   new_lhs)
+      new_lhs = _reshape_axis_into(lhs_spec[0], lhs_spec[0], new_lhs)
+      out = conv_general_dilated(new_lhs, rhs, window_strides, padding,
+                                 lhs_dilation, rhs_dilation, dimension_numbers,
+                                 feature_group_count, batch_group_count,
+                                 precision=precision)
+      out = _reshape_axis_out_of(out_spec[0], lhs.shape[lhs_bdim], out)
+      return out, out_spec[0]
 
   elif rhs_bdim is not None:
-    if feature_group_count == 1:
+    if feature_group_count == 1 and batch_group_count == 1:
       new_rhs = _reshape_axis_into(rhs_bdim, rhs_spec[0], rhs)
       out = conv_general_dilated(lhs, new_rhs, window_strides, padding,
-                                lhs_dilation, rhs_dilation, dimension_numbers,
-                                feature_group_count, precision=precision)
+                                 lhs_dilation, rhs_dilation, dimension_numbers,
+                                 feature_group_count, batch_group_count,
+                                 precision=precision)
       out = _reshape_axis_out_of(out_spec[1], rhs.shape[rhs_bdim], out)
       return out, out_spec[1]
     else:
-      # feature_group needs to be outermost, so we need to factor it out of the
+      # groups needs to be outermost, so we need to factor it out of the
```

Comment:
I might be missing the context but to make the plural subject agree with the rest of the sentence maybe we should do

need -> needs
it -> them

Reply:
Yes I just got lazy editing the existing text. The logic is the same for `batch_group_count` and `feature_group_count` in this case.